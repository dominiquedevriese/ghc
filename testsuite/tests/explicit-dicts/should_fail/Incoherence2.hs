module Incoherence2 where

import GHC.Classes (Ord(..))

foo :: (Ord a, Eq a) => a -> a
foo = undefined

bar :: Eq a => Ord.Dict a -> a -> a
bar dict = foo ((dict :: Ord a))
