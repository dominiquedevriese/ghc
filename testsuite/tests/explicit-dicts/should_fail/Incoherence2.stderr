
Incoherence2.hs:9:12: error: [-Wincoherence (in -Wdefault), -Werror=incoherence]
    • Explicit dictionary application to: ‘Ord a’
      is not allowed because of possible incoherence:
        Through an ancestor: ‘Eq a’ occurs in the context
    • In the expression: foo ((dict :: Ord a))
      In an equation for ‘bar’: bar dict = foo ((dict :: Ord a))
